### PR TITLE
Add role to admin/counselor enrollment form

### PR DESCRIPTION
**Pull request by megcrenshaw**
Adds role to admin/counselor on enrollment form. The `role` is already a [serialized attribute in the enrollment model](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/enrollment.rb#L76), so on the backend, the enrollment looks like
```
=> #<Pd::Enrollment id: 39, ..., properties: {"role"=>"Administrator", "grades_teaching"=>["Grade 9-12"]}, application_id: nil>
```

The dropdown:
Screenshot 2023-01-09 at 3 16 15 PM

When you select 'Other', a text box appears prompting for more info:
Screenshot 2023-01-09 at 3 16 32 PM

## Links

- Jira Ticket: https://codedotorg.atlassian.net/browse/ACQ-327

## Testing story

Added unit tests, and tested locally by adding workshops at http://localhost-studio.code.org:3000/pd/workshop_dashboard

## Follow-up work

We have a ticket to add UI tests for these enrollment forms.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by dmcavoy** [comment]
Hey @megcrenshaw - Are you planning to add the regional partner information based on the school chosen in a later PR?

**Comment by megcrenshaw** [comment]
Good call Dani –– I've gone ahead and made a separate ticket for that to make it easier to track!

File: apps/test/unit/code-studio/pd/workshop_enrollment/enroll_formTest.js [hunk 1]
```diff
@@ -236,6 +236,39 @@ describe('Enroll Form', () => {
...
+
+    it('does not display describe role question after answered as counselor or admin', () => {
+      enrollForm.setState({role: 'Administrator'});
+      expect(enrollForm.find('#describe_role')).to.have.length(0);
+
+      enrollForm.setState({role: 'Counselor'});
+      expect(enrollForm.find('#describe_role')).to.have.length(0);
+    });
+  });
+
```
Comment (TurnerRiley) [inline apps/test/unit/code-studio/pd/workshop_enrollment/enroll_formTest.js:271]: Thanks for adding these tests !

File: apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx [hunk 2]
```diff
@@ -378,6 +381,10 @@ export default class EnrollForm extends React.Component {
       'I don’t have experience teaching any of these courses'
     ]);
 
+    const roles =
+      (this.props.workshop_course === CSF && CSF_ROLES) ||
+      (this.props.workshop_course === ADMINCOUNSELOR && ADMIN_COUNSELOR_ROLES);
+
```
Comment (TurnerRiley) [inline apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx:387]: Super readable and clear solution!

**Review by TurnerRiley — APPROVED** [review]

**Review by davidsbailey — APPROVED** [review]
Looks good!

**Pull request merged**